src/main/java/com/zaxxer/sansorm/internal/OrmWriter.java

```diff
@@ -68,12 +68,11 @@ protected boolean removeEldestEntry(Map.Entry<Introspected, String> eldest)
          throw new RuntimeException("insertListBatched() is not supported for objects with self-referencing columns due to Derby limitations");
       }
 
-      String[] columnNames = introspected.getInsertableColumns();
-
-      try (PreparedStatement stmt = createStatementForInsert(connection, introspected, columnNames)) {
+      FieldColumnInfo[] insertableFcInfos = introspected.getInsertableFcInfos();
+      try (PreparedStatement stmt = createStatementForInsert(connection, introspected, insertableFcInfos)) {
          int[] parameterTypes = getParameterTypes(stmt);
          for (T item : iterable) {
-            setStatementParameters(item, introspected, columnNames, hasSelfJoinColumn, stmt, parameterTypes, null);
+            setStatementParameters(item, introspected, insertableFcInfos, stmt, parameterTypes, null);
```

Comment:
hasSelfJoinColumn is redundant. This information is contained in both introspected and insertableFcInfos.